src/images.py

```diff
@@ -0,0 +1,177 @@
+"""Module for matching individual images from raw and corrected directories
+and computing intensity slopes from the foreground and background regions."""
+import warnings
+
+from pathlib import Path
+
+import matplotlib.pyplot as plt
+import numpy as np
+
+def match_image_pairs(raw_dir, corrected_dir, key_parts=4, extensions=None):
+    """
+    Match images from two directories based on a common key in their filenames.
+
+    Parameters
+    ----------
+    raw_dir : str or Path
+        Directory containing raw images.
+    corrected_dir : str or Path
+        Directory containing corrected images.
+    key_parts : int
+        Number of underscore-separated parts of the filename to use as the key.
+
+        For example, if filenames are named as "AS_1234_A01f00d0.tif", you should
+        set key_parts=3, as the key will be "AS_1234_A01f00d0". The key must include
+        well, site, and channel information to ensure correct pairing.
+    extensions : list or None
+        List of file extensions to include (e.g., ['.tif', '.TIFF']). Default extensions
+        include only versions of TIFF extensions.
+
+    Returns
+    -------
+    List[Tuple[Path, Path]]
+        List of matched (raw_image, corrected_image) pairs.
+    """
+    raw_dir = Path(raw_dir)
+    corrected_dir = Path(corrected_dir)
+    extensions = extensions or [".tif", ".tiff", ".TIFF"]
```

Comment:
Consider using list reference as a default for the function definition.

Reply:
Updated to be its own helper function